Python/Product/PythonTools/PythonTools/Project/CondaEnvCreateInfoBar.cs

```diff
@@ -27,51 +29,62 @@
 
 namespace Microsoft.PythonTools.Project {
     internal sealed class CondaEnvCreateInfoBar : PythonProjectInfoBar {
-        public CondaEnvCreateInfoBar(PythonProjectNode projectNode)
-            : base(projectNode) {
+        public CondaEnvCreateInfoBar(IServiceProvider site, PythonProjectNode projectNode, IPythonWorkspaceContext workspace)
+            : base(site, projectNode, workspace) {
         }
 
         public override async Task CheckAsync() {
             if (IsCreated) {
                 return;
             }
 
-            if (!Project.Site.GetPythonToolsService().GeneralOptions.PromptForEnvCreate) {
+            var projectOrWorkspaceName = Project?.Caption ?? Workspace?.WorkspaceName ?? string.Empty;
+
+            if (!Site.GetPythonToolsService().GeneralOptions.PromptForEnvCreate) {
                 return;
             }
 
-            var suppressProp = Project.GetProjectProperty(PythonConstants.SuppressEnvironmentCreationPrompt);
-            if (suppressProp.IsTrue()) {
+            if (IsSuppressed(PythonConstants.SuppressEnvironmentCreationPrompt)) {
                 return;
             }
 
             // Skip if active is already conda
-            var active = Project.ActiveInterpreter;
+            var active = Project?.ActiveInterpreter ?? Workspace?.CurrentFactory;
             if (IsCondaEnvOrAnaconda(active) && active.IsRunnable()) {
                 return;
             }
 
-            var yamlPath = Project.GetEnvironmentYmlPath();
-
-            var all = Project.InterpreterFactories.ToArray();
-            var allConda = all.Where(IsCondaEnvOrAnaconda).ToArray();
-            var foundConda = allConda.Where(f => f.IsRunnable()).ToArray();
-            var condaNotFoundNames = Project.InvalidInterpreterIds
-                .Where(id => CondaEnvironmentFactoryProvider.IsCondaEnv(id))
-                .Select(id => CondaEnvironmentFactoryProvider.NameFromId(id))
-                .Where(name => name != null)
-                .ToArray();
-
-            string existingName;
-            if (condaNotFoundNames.Any() && !foundConda.Any()) {
-                // Propose to recreate one of the conda references, since they are all missing
-                existingName = condaNotFoundNames.First();
-            } else if (!foundConda.Any() && !string.IsNullOrEmpty(yamlPath)) {
-                // Propose to create a new one, since there's a yaml file and no conda references
-                existingName = null;
-            } else {
-                // Nothing to do
-                return;
+            var yamlPath = Project?.GetEnvironmentYmlPath() ?? Workspace?.GetEnvironmentYmlPath();
+            string existingName = null;
```

Comment:
if(Project != null) is false, then this string will remain null and may cause problems in "AddEnvironmentDialog.ShowAddCondaEnvironmentDialogAsync"